src/algorithms/diamondsquare.jl

```diff
@@ -0,0 +1,269 @@
+which varies from 0.0 (low autocorrelation) to 1.0 (high autocorrelation) --- 
+note this is non-inclusive and H = 0 and H = 1 will not behave as expected.
+The result of the diamond-square algorithm is a fractal with dimension D = 2 + H.
+
+A similar algorithm, midpoint-displacement, functions almost
+identically, except that in DiamondSquare, the square step interpolates
+edge midpoints from the nearest two corners and the square's center, where as 
+midpoint-displacement only intepolates from the nearest corners (see `MidpointDisplacement`).
+
+"""
+struct DiamondSquare <: NeutralLandscapeMaker
+    H::Float64
+    function DiamondSquare(H::T) where {T <: Real}
+        @assert 0 <=  H < 1
+        new(H)
+    end
+end
+
+"""
+    DiamondSquare() 
+
+    Initializes DiamondSqaure with H = 0.5.
+"""
+DiamondSquare() = DiamondSquare(0.5)
+
+"""
+    MidpointDisplacement()
+
+    Creates a midpoint-displacement algorithm object `MidpointDisplacement`. The degree of spatial autocorrelation is controlled by a parameter `H`,
+    which varies from 0.0 (low autocorrelation) to 1.0 (high autocorrelation) --- note this is non-inclusive and H = 0 and H = 1 will not behavive as expected.
+
+    A similar algorithm, diamond-square, functions almost
+    identically, except that in diamond-square, the square step interpolates
+    edge midpoints from the nearest two corners and the square's center, where as 
+    `MidpointDisplacement` only intepolates from the nearest corners (see `DiamondSquare`).
+"""
+struct MidpointDisplacement <: NeutralLandscapeMaker
+    H::Float64
+    function MidpointDisplacement(H::T) where {T <: Real}
+        @assert 0 <= H < 1
+        new(H)
+    end
+end
+MidpointDisplacement() = MidpointDisplacement(0.5)
+
+"""
+    _landscape!(mat, alg::Union{DiamondSquare, MidpointDisplacement}; kw...)
+
+    Check if `mat` is the right size.
+    If mat is not the correct size (DiamondSquare can only run on a lattice of size NxN where N = (2^n)+1 for integer n),
+    allocates the smallest lattice large enough to contain `mat` that can run DiamondSquare.
+"""
+function _landscape!(mat, alg::Union{DiamondSquare, MidpointDisplacement}; kw...) where {IT <: Integer}
+
+    rightSize::Bool = _isPowerOfTwo(size(mat)[1]-1) && _isPowerOfTwo(size(mat)[2]-1)
+    latticeSize::Int = size(mat)[1]
+
+    dsMat = mat
+    if !rightSize
+        dim1, dim2 = size(mat)
+        smallestContainingLattice::Int = 2^ceil(log2(max(dim1, dim2))) + 1
+        @warn "$alg cannot be run on the input dimensions ($dim1 x $dim2),
+                and will instead run on the next smallest valid size ($smallestContainingLattice x $smallestContainingLattice).
+                This can slow performance as it involves additional memory allocation."
+        dsMat = zeros(smallestContainingLattice, smallestContainingLattice)
+    end
+    _diamondsquare!(dsMat, alg)
+
+    mat .= dsMat[1:size(mat)[1], 1:size(mat)[2]]
```

Comment:
Should this maybe use the center rectangle rather than top-right one? Note sure if it makes a difference. @gottacatchenall